python/sg_qwidgets/__init__.py

```diff
@@ -27,6 +27,8 @@
     SGQMenu,
     SGQProgressBar,
     SGQPushButton,
+    SGSubmitPushButton,
+    SGCancelPushButton,
```

Comment:
Minor question: why do we define so many things in our `__init__.py`? Would that not be easier (and less code to maintain) to keep it (almost) empty and use the regular Python import mechanism to import from sg_qwidgets?


This comment/question is not specific to this change but more like a general question of mine of the SGTK code. Like is there something I'm missing?

Reply:
Good question. I'm not entirely sure why this was the approach taken when the framework was initially set up, when implementing this `sg_qwidgets` modules I just followed the other modules.

One reason that defining each class is beneficial is that you can restrict which classes the consumer has access to. If the framework is kind of trying to act like an "api", this makes it more clear which modules the consumer should use.

The framework modules can be imported by a Toolkit EAF using this method:
```
sg_qwidgets = sgtk.platform.current_bundle().import_module("sg_qwidgets")
```
, through this import mechanism the sg_qwidgets/__ init __.py cannot be empty, it must import any modules that you would want to access by:
```
sg_qwidgets.SGQPushButton
```

You could replace all the individual imports with a single line:
```
from .sg_qwidgets import *
```
but then as noted above, a consumer has access to all modules defined and imported in the `sg_qwidgets` file, e.g. QtCore, QtGui, etc.

This example is for the `sg_qicons` module, if we had imported like `from .sg_qicons import *`

![image](https://github.com/user-attachments/assets/cd77afed-6dec-4363-bce2-029ccf4ddf61)